From 8e6c227f5d427aa1409515d1804a5c39de5995f8 Mon Sep 17 00:00:00 2001
From: Mykola Lysenko <Mykola.Lysenko@amd.com>
Date: Mon, 14 Dec 2015 19:22:38 +0800
Subject: [PATCH 0617/1110] drm/amd/dal: guard drm mst manager usage

In case mst branch got disconnected, and mst manager
is not active, we should not call its functions.

Signed-off-by: Mykola Lysenko <Mykola.Lysenko@amd.com>
Signed-off-by: Harry Wentland <harry.wentland@amd.com>
Acked-by: Harry Wentland<harry.wentland@amd.com>
---
 drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dc_helpers.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dc_helpers.c b/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dc_helpers.c
index 0230ee9..ddc40c9 100644
--- a/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dc_helpers.c
+++ b/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dc_helpers.c
@@ -192,6 +192,10 @@ bool dc_helpers_dp_mst_write_payload_allocation_table(
 		return false;
 
 	mst_mgr = &aconnector->mst_port->mst_mgr;
+
+	if (!mst_mgr->mst_state)
+		return false;
+
 	mst_port = aconnector->port;
 
 	if (enable) {
@@ -379,6 +383,9 @@ bool dc_helpers_dp_mst_poll_for_allocation_change_trigger(
 
 	mst_mgr = &aconnector->mst_port->mst_mgr;
 
+	if (!mst_mgr->mst_state)
+		return false;
+
 	ret = drm_dp_check_act_status(mst_mgr);
 
 	if (ret)
@@ -408,6 +415,9 @@ bool dc_helpers_dp_mst_send_payload_allocation(
 
 	mst_mgr = &aconnector->mst_port->mst_mgr;
 
+	if (!mst_mgr->mst_state)
+		return false;
+
 	ret = drm_dp_update_payload_part2(mst_mgr);
 
 	if (ret)
-- 
2.7.4

